### PR TITLE
Apply the RAII idiom in Chat classes.

### PR DESCRIPTION
**Pull request by djarek**
Also reformatted the ChatChannel constructor for better readability.

### CODE REVIEW

**Comment by HeavenIsLost** [comment]
You pretend to do that on the whole tfs? danm, it will break a lot of my code i guess(tfs 1.1).

File: src/chat.h [hunk 1]
```diff
@@ -152,13 +159,13 @@ class Chat
...
-		std::map<Party*, ChatChannel*> partyChannels;
-		std::map<uint32_t, ChatChannel*> guildChannels;
+		std::map<uint16_t, PrivateChatChannel> privateChannels;
+		std::map<Party*, ChatChannel> partyChannels;
+		std::map<uint32_t, ChatChannel> guildChannels;
 
 		LuaScriptInterface m_scriptInterface;
 
-		ChatChannel* dummyPrivate;
+		std::unique_ptr<PrivateChatChannel> dummyPrivate;
```
Comment (kornholi) [inline src/chat.h]: does this actually need to be a pointer?

Comment (djarek) [inline src/chat.h]: Actually, it doesn't have to. I'll embed that variable directly into the Chat class.

**Comment by djarek** [comment]
@HeavenIsLost 
These really are low-hanging fruit. Applying RAII to the map code or core game logic is a little out of my reach at the moment - there's way too much spaghetti in there for my taste. Unless you had some customized chat features this shouldn't break any lua scripts or code outside the chat files.

**Comment by marksamman** [comment]
Thanks!

**Pull request merged**